### DISCUSSION TITLE
Athom 4 Gang Touch Switch Update to MEPLHAA Script issue

### DISCUSSION THREAD

**Discussion by jaysonpriest** [RavenSystem/esp-homekit-devices#2034] [MEPLHAA Scripts]
Hi All, 

I have both 2 Gang, 3Gang and 4 Gang Athom switches that I have been using for a couple of years. I had been holding off updating to 12.0 due to the breakage change with IO configuration. I have recently undertook the update and have configured successfully the 2 Gang and 3 Gang without issues. I also thought, at first, that I had the 4 gang version working correctly, but found a peculiar issue. The unit works well from homekit (responds by correctly switching relays and turning on appropriate buttons. However, I then realised a little later that the 4th button after a period of ~10 seconds does not respond to touch...however, if one of the other buttons is toggled it will then respond correctly...but after 10 seconds it stops again until hitting one of the other buttons again.

The Athom switch is this one https://templates.blakadder.com/athom_SW03-TAS-4US.html and my Script looks like this:

{"c":{"io":[[[4,13,15,14,0],2],[[3,5,12,16],6]],"l":0,"i":1,"b":[{"g":0,"t":5}],"n":"Bathroom2-s1"},"a":[
{"0":{"r":[[13]]},"1":{"r":[[13,1]]},"b":[[12]],"s":5},
{"0":{"r":[[4]]},"1":{"r":[[4,1]]},"b":[[3]],"s":5},
{"0":{"r":[[15]]},"1":{"r":[[15,1]]},"b":[[5]],"s":5},
{"0":{"r":[[14]]},"1":{"r":[[14,1]]},"b":[[16]],"s":5}]}

If anyone can point at an obvious error or potential issues I would be very appreciative. Just for the record, prior to the updgrade these performed as expected. 

Cheers

**Comment by RavenSystem** [comment]
You must enable [Binary Input Continuous Mode](https://github.com/RavenSystem/esp-homekit-devices/wiki/General-Configuration#Binary-Input-Continuous-Mode) needed by GPIO 16 button.

{"c":{"io":[[[4,13,15,14,0],2],[[3,5,12,16],6]],"l":0,"c":1,"b":[[0,5]],"n":"Bathroom2-s1"},"a":[
{"0":{"r":[[13]]},"1":{"r":[[13,1]]},"b":[[12]],"s":5},
{"0":{"r":[[4]]},"1":{"r":[[4,1]]},"b":[[3]],"s":5},
{"0":{"r":[[15]]},"1":{"r":[[15,1]]},"b":[[5]],"s":5},
{"0":{"r":[[14]]},"1":{"r":[[14,1]]},"b":[[16]],"s":5}]}

**Reply by jaysonpriest** [reply]
Jose... thank you so much for the quick reply...I can't believe i missed that. I can confirm that that works perfectly.

**Reply by RavenSystem** [reply]
You are welcome. You have done a great work migrating all your scripts to new version.
